setup.py

```diff
@@ -14,6 +15,16 @@
 test_requirements = [
 ]
 
+INSTALL_REQUIRES = [
+    'w3lib>=1.8.0',
+    'lxml>=2.3',
+    'six>=1.5.2',
+    'cssselect>=0.9'
+]
+
+if sys.version_info[0:2] < (3, 0):
```

Comment:
I think that this will not do what we want in all cases, because parsel uses universal wheel (https://github.com/scrapy/parsel/blob/master/setup.cfg).

First, how to check (it's important to do it for this PR anyway):

- build a source and a universal wheel, the command is ``python setup.py sdist bdist_wheel`` (this will generate two files)
- make fresh virtual environments for python 2 and python 3 with recent pip, and try installing both files (also in separate virtualenvs): this should result in functools32 being installed only for python2

I didn't check, but I think that for a wheel, it will results in functools32 not installed in both cases (if you build the wheel under python 3), or installed in both cases, if you build a wheel under python 2, which is done for pypi releases: https://github.com/scrapy/parsel/blob/f6103c8808170546ecf046b9f4ea6dead94de189/.travis.yml#L59
The reason is that setup.py code is not executed for a wheel, but this is the most common (and fastest) way of installation.

In order to fix this, a conditional dependency must be added in a different way, like it's added for pypy in scrapy (https://github.com/scrapy/scrapy/blob/master/setup.py) - but in this case, we want to condition on python version, I hope there are such examples.

If anything is not clear, don't hesitate to ask - and also please check that there is indeed a problem before fixing.

Reply:
I checked and indeed theres some errors:
Wheel created with **Python2**: 
- **Python 2**, install with whl: Yes, functools32 was installed. No problem with it!
- **python 3**, install with whl: It was trying to install functools32 and failed to install the package.

Wheel created with **python3**:
- **python3**, install with whl: no problem with the installation, it didn't try to install functools32
- **python2**, install with whl: functools32 was not installed